connect/rest-extension/src/main/java/org/apache/kafka/connect/rest/extension/BasicAuthSecurityRestExtension.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.connect.rest.extension;
+
+import org.apache.kafka.common.utils.AppInfoParser;
+import org.apache.kafka.connect.rest.ConnectRestExtension;
+import org.apache.kafka.connect.rest.ConnectRestExtensionContext;
+
+import java.io.IOException;
+import java.util.Map;
+
+/**
+ * {@link BasicAuthSecurityRestExtension} provides the ability to authenticate incoming BasicAuth
```

Comment:
Is this the only REST extension that we'll have in this project? If so, should we have a better and more descriptive name for the project rather than simple `rest-extension`?

Reply:
ATM, this is the only implementation. I named it just like other components for transforms. I'm not too particular about the module name being generic. I could call it connect-basic-auth-extension

Reply:
Yeah, maybe `basic-auth-extension` (e.g., to go with `file`, etc.). I'm just concerned that `rest-extension` is pretty generic and actually sounds like it's the API, not a reference impl.